lib/services/role.go

```diff
@@ -262,18 +262,40 @@ func (r *RoleV2) SetForwardAgent(forwardAgent bool) {
 
 // Check checks validity of all parameters and sets defaults
 func (r *RoleV2) CheckAndSetDefaults() error {
+	// make sure we have defaults for all fields
 	if r.Metadata.Name == "" {
 		return trace.BadParameter("missing parameter Name")
 	}
+	if r.Metadata.Namespace == "" {
```

Comment:
one thing that worry me a bit - it will be helpful to distinguish between "not specified" and left empty, e.g your pull request does not allow me to prohibit access to all nodes at all, it will allways override with wildcard.